python/lsst/jointcal/jointcal.py

```diff
@@ -264,6 +264,11 @@ class JointcalConfig(pexConfig.Config):
         dtype=float,
         default=5.0,
     )
+    failOnChi2Increase = pexConfig.Field(
+        doc="Ratio of old/new chi2 to fail the fit on (large chi2 increases are generally non-recoverable.",
```

Comment:
First, a nit - the parentheses are unterminated:

```suggestion
        doc="Ratio of old/new chi2 to fail the fit on (large chi2 increases are generally non-recoverable).",
```

Second, a suggestion: I think this should be named `maxChi2Increase` or something similar to convey that it's a maximum which should not be breached.

Third, a design question: what is the tradeoff between high and low values here? Documenting that can convey a lot to the user. Like, gzip says you can run it with `gzip -#`, where lower numbers give you faster execution time, but worse compression ratios; only the user really knows which is appropriate, so we handle that with run-time configuration.

If there is no clear tradeoff at all, maybe this should be a constant, and it's an invariant you assert against. That seems like it _might_ be the case here - you might actually want to just assert that `old/new < 1e4` or something, and not burden the user with figuring this out. Each configurable option is easy to add but bears cost forever for users who have to figure out whether it's significant to them.

Reply:
The tradeoff is whether the algorithm will 1) continue to run at all (and, based on the samples of processing runs I've looked at, continue to wander off into more and more terrible chi2s) which can take quite a long time until it exceeds `max_steps` iterations, or 2) fail early with a message, allowing someone to dig into trying to figure out why chi2 increased. My thought in making it configurable was that someone might want to try to let it run to completion. The better choice is probably for them to add debug output and re-run to get more information about how the fit is failing early on.

I've revised the raised error to this effect.